### PR TITLE
#166790110: Super admin should be able to create users

### PR DESCRIPTION
**Pull request by kagabof**
#### What does this PR do?

- Add admin functionalities.

#### Description of Task to be completed?

- Added admin controller with crud functionality
- Added test for the admin functionality.
- Added routers for the admin functionality,
- Added swagger documentation for admin functionality.

#### How should this be manually tested?

- By using postman: 
     - PATCH/  http://127.0.0.1:3000/api/adminuser/:id
     - POST/  http://127.0.0.1:3000/api/adminuser
     - DELETE/ http://127.0.0.1:3000/api/adminuser/:id
     - GET/ http://127.0.0.1:3000/api/adminuser

- By using swagger documentation.
-By running test: npm run test.

#### Any background context you want to provide?
- NONE
#### What are the relevant pivotal tracker stories?

[#166790110](https://www.pivotaltracker.com/story/show/166790110)

#### Screenshots

Screen Shot 2019-07-29 at 21 39 50

#### Questions:

- NONE

### CODE REVIEW

File: src/controllers/adminContoller.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/* eslint-disable no-unused-expressions */
```
Comment (FlevianK) [inline src/controllers/adminContoller.js]: Super admin should not be created, update or deleted, should be seeded in the database at the time of up deployment. super admin is a user who has higher permission. For creating, deleting, updating or fetching should uses methods in the user controller, just modify those functions a little bit to ensure that admin can also perform those operations on any user.

File: src/controllers/helpers/findUser.js [hunk 2]
```diff
@@ -8,4 +8,20 @@ const findUser = async (id) => {
   return dataValues || {};
 };
 
-export default findUser;
+const findUserData = async (object) => {
+  const user = await User.findOne(object);
+  return user;
+};
+
+const findUserExist = async (username, email) => {
```
Comment (FlevianK) [inline src/controllers/helpers/findUser.js:10]: I don't understand how this function works

File: src/routes/adminRoute.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import express from 'express';
+import Auth from '../middlewares/Auth';
+import checkAction from '../middlewares/roleCheck';
+import adminController from '../controllers/adminContoller';
+
+const admin = express.Router();
+
+const { verifyToken } = Auth;
+
+admin.post('/adminuser', verifyToken, checkAction('User'), adminController.createUser);
```
Comment (FlevianK) [inline src/routes/adminRoute.js]: You actually don't need this routes, admin is a user hence user's routes are enough for any admin access and operation

**Review by FlevianK — CHANGES_REQUESTED** [review]
Admin is user the only difference is that he/she can access any other user and perform operations on them as if he/she is the other logged in user.

Kindly implement the reviews.

File: src/routes/index.js [hunk 4]
```diff
@@ -46,13 +48,13 @@ router.post('/api/users/passwordreset/:token', decodeResetPasswordToken, checkEm
...
 router.get('/api/allusers/', verifyToken, UserProfile.getAllUser);
 router.post('/api/users/login', signinValidation, AuthController.signin);
 router.patch('/api/users/verification/:userToken', UserController.verifyUser);
 
-router.patch('/api/users/:username', verifyToken, bodyValidation, usernameAvailability, UserProfile.updateProfile);
-router.get('/api/users/', verifyToken, UserProfile.getAllUser);
+router.patch('/api/users', verifyToken, UserProfile.updateProfile);
+router.get('/api/users', verifyToken, UserProfile.getAllUser);
 
+router.delete('/api/users/:id', verifyToken, checkAction('User'), UserController.deleteUser);
```
Comment (FlevianK) [inline src/routes/index.js:58]: Why are you having this action `checkAction('User')` here and not at the controller level?

File: src/controllers/userProfile.js [hunk 5]
```diff
@@ -31,20 +32,32 @@ class UserProfile {
     * @return {object} returns an object containing the updated user profile
     */
   static async updateProfile(req, res) {
```
Comment (FlevianK) [inline src/controllers/userProfile.js:34]: This should be refactored better during code maintainability session

**Review by FlevianK — APPROVED** [review]
Implement review that relates to routing

File: src/routes/index.js (hunk 4 above)
Comment (kagabof) [reply src/routes/index.js:58]: `checkAction` is a middleware function that checks if a user trying to access the router has the priority of deleting in a user table.

**Pull request merged**